bin/install_swiftpm_dependencies

```diff
@@ -1,9 +1,67 @@
 #!/bin/bash -eu
 
+CUSTOM_PACKAGE_RESOLVED_PATH=${1-}
+BUILD_TYPE=${2-}
+SPM_CACHE_LOCATION="$HOME/spm"
+
+WORKSPACE_PACKAGE_RESOLVED_PATH=(*.xcworkspace/xcshareddata/swiftpm/Package.resolved)
+ROOT_PACKAGE_RESOLVED_PATH=Package.resolved
+
+# Find where Package.resolved is located
+if [[ -n "$CUSTOM_PACKAGE_RESOLVED_PATH" ]]; then
+  PACKAGE_RESOLVED_LOCATION=$CUSTOM_PACKAGE_RESOLVED_PATH
+elif  [[ -f ${WORKSPACE_PACKAGE_RESOLVED_PATH[0]} ]]; then
+  BUILD_TYPE="XCODEBUILD"
+  PACKAGE_RESOLVED_LOCATION=${WORKSPACE_PACKAGE_RESOLVED_PATH[0]}
+elif  [[ -f $ROOT_PACKAGE_RESOLVED_PATH ]]; then
+  BUILD_TYPE="SWIFT"
+  PACKAGE_RESOLVED_LOCATION=$ROOT_PACKAGE_RESOLVED_PATH
+else
+  echo "Unable to find Package.resolved"
+fi
```

Comment:
I'm surprised that `shellcheck` didn't make you add quotes around all of those to prevent issues if paths have spaces? 🤔 

```suggestion
if [[ -n "$CUSTOM_PACKAGE_RESOLVED_PATH" ]]; then
  PACKAGE_RESOLVED_LOCATION="$CUSTOM_PACKAGE_RESOLVED_PATH"
elif  [[ -f "${WORKSPACE_PACKAGE_RESOLVED_PATH[0]}" ]]; then
  BUILD_TYPE="XCODEBUILD"
  PACKAGE_RESOLVED_LOCATION="${WORKSPACE_PACKAGE_RESOLVED_PATH[0]}"
elif  [[ -f "$ROOT_PACKAGE_RESOLVED_PATH" ]]; then
  BUILD_TYPE="SWIFT"
  PACKAGE_RESOLVED_LOCATION="$ROOT_PACKAGE_RESOLVED_PATH"
else
  echo "Unable to find Package.resolved"
fi
```